assets/js/controllers/datepicker_controller.js

```diff
@@ -0,0 +1,32 @@
+/*!
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { Controller } from '@hotwired/stimulus';
+import { TempusDominus } from '@eonasdan/tempus-dominus';
+
+/* stimulusFetch: 'lazy' */
+export default class extends Controller {
```

Comment:
Declaring the date picker like this is a huge huge improvement.

But needs the admin to load assets via the webpack encore bundle functions to achieve all the benefits.

With this:
1. TempusDominus only gets loaded if there is a date picker on the page (this can only be done with webpack-encore-bundle).
3. It dispatches events that can be captured via javascript or other stimulus controllers
4. It works will all elements added after page load

Reply:
It can also be easily tested, which is also great for our javascript ecosystem. This can be the first piece of rewriting all the Js we have out there.

I could even convert it to typescript.

wdyt @VincentLanglet ?

Reply:
With the current code you can try it out adding this to javascript loaded by admin-bundle:

```
                        ->arrayNode('javascripts')
                            ->defaultValue([
                                'bundles/sonataadmin/app.js',
                                'bundles/sonataform/runtime.js',
                                'bundles/sonataform/192.js',
                                'bundles/sonataform/app.js',
                            ])
                            ->prototype('scalar')->end()
                        ->end()
```

The ideal would be to only load "app" with the webpack encore bundle function and it handles everything.

Reply:
> It can also be easily tested, which is also great for our javascript ecosystem. This can be the first piece of rewriting all the Js we have out there.
> 
> I could even convert it to typescript.
> 
> wdyt @VincentLanglet ?

This seems nice.

```
                        ->arrayNode('javascripts')
                            ->defaultValue([
                                'bundles/sonataadmin/app.js',
                                'bundles/sonataform/runtime.js',
                                'bundles/sonataform/192.js',
                                'bundles/sonataform/app.js',
                            ])
                            ->prototype('scalar')->end()
                        ->end()
```
It would be better indeed to have only one sonataform file to load.

Reply:
This is no longer an issue with https://github.com/sonata-project/SonataAdminBundle/pull/7838.